src/controllers/EvidenceDownloadController.ts

```diff
@@ -34,21 +46,39 @@ export class EvidenceDownloadController extends BaseAsyncHttpController {
             return this.renderDownloadError();
         }
 
-        let readable: Readable;
+        return await this.downloadAttachment(metadata, this.httpContext.response);
+    }
+
+    private async downloadAttachment(metadata: FileMetadata, res: Response): Promise<void> {
+
         try {
-            readable = await this.fileTransferService.download(fileId);
+            const downloadStream: Readable = await this.fileTransferService.download(metadata.id);
+            res.setHeader('content-disposition', `attachment; filename=${metadata.name}`);
+            return this.pipeDataIntoStream(downloadStream, res);
+
         } catch (err) {
+
             if (err instanceof FileNotReadyError) {
                 return this.renderDownloadError();
             } else {
                 throw err;
             }
         }
+    }
 
-        this.httpContext.response.setHeader('content-disposition', `attachment; filename=${metadata.name}`);
-        return await this.pipeDataIntoStream(readable, this.httpContext.response);
+    private getDownloadPath(req: Request, fileId: string): string {
+
+        const appealId = req.query[APPEAL_ID_QUERY_KEY];
+        const companyNumber = req.query[COMPANY_NUMBER_QUERY_KEY];
+
+        if (!appealId || !companyNumber) {
+            throw Error('Expected appealId and company number as query parameters but none found');
```

Comment:
Should this error message be: `'Expected **either** appealId **or** company number as query parameters but none found'`

Reply:
or should this check `(!appealId && !companyNumber)`?

Reply:
either not being present is correct